### ISSUE TITLE
Input goes through even though key is binded, only with --input-device option.

### ISSUE THREAD

**Issue by IJR222** [swaywm/sway#6634]
### Please fill out the following:
- **Sway Version:**
  - `swaymsg -t get_version` or `sway -v`
sway version 1.6.1
- **Debug Log:**
  - Run `sway -d 2> ~/sway.log` from a TTY and upload it to a pastebin, such as gist.github.com.
  - This will record information about sway's activity. Please try to keep the reproduction as brief as possible and exit sway.
  - Attach the **full** file, do not truncate it.
https://gist.github.com/IJR222/784a30fa7bb2912de9d5970a3cae423e
- **Configuration File:**
  - Please try to produce with the default configuration.
  - If you cannot reproduce with the default configuration, please try to find the minimal configuration to reproduce.
  - Upload the config to a pastebin such as gist.github.com.
Reproduced with default config

- **Description:**
  - The steps you took in plain English to reproduce the problem.

Sway normally stops keys from being forwarded to applications when they are bound to an action. Binding my keyboard with "swaymsg 'bindsym --input-device='1008:1610:Primax_HP_USB_Keyboard' f 'exec firefox''" still passes the keys through to all applications.

**Comment by arnej** [comment]
I'm currently building a keypad for extra macro buttons and tried to use bindsym with --input-device to run the commands only when the keypad sent them.

This works, except that the pressed key is also passed through to the active application as described in this issue.

Do you maybe have a hint what part of the code is relevant for this so that I can check what could cause this behavior?
